### PR TITLE
Turn Band-Selection Code into Library

### LINKED ISSUE #121: Bring Band-Selection Code into the Library

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

The model refinement and band-selection code found in `band_selection.ipynb` has been turned into a library.

TODO
   - [x] Code to produce (sample) PyTorch model
   - [x] Band-selection code
   - [x] Test library
   - [x] Ensure that `np.array` and `torch.tensor` dimensions given in the comments are correct
 
Closes #121 

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] README.md updated if necessary to reflect the changes

### CODE REVIEW

**Comment by jamesmcclain** [comment]
@jpolchlo Ready for review

File: src/hyperspectral/hyperspectral/band_selection/models.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+# LIABILITY, OR  TORT (INCLUDING  NEGLIGENCE OR OTHERWISE)  ARISING IN
+# ANY WAY  OUT OF  THE USE OF  THIS SOFTWARE, EVEN  IF ADVISED  OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import torch as torch
+import torch.nn.functional as F
+
+
+def whitening_matrix(m: np.array):
```
Comment (jpolchlo) [inline src/hyperspectral/hyperspectral/band_selection/models.py:36]: Note that there is an existing (but not as robust) definition of this idea [here](https://github.com/azavea/nasa-hyperspectral/blob/master/src/hyperspectral/hyperspectral/math/rpca.py#L136-L137).  Please consider moving this implementation to replace that one.

File: src/hyperspectral/hyperspectral/band_selection/train.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# LIABILITY, OR  TORT (INCLUDING  NEGLIGENCE OR OTHERWISE)  ARISING IN
+# ANY WAY  OUT OF  THE USE OF  THIS SOFTWARE, EVEN  IF ADVISED  OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import torch as torch
+import torch.nn.functional as F
+
+
+def vanilla_train(model: torch.nn.Module,
```
Comment (jpolchlo) [inline src/hyperspectral/hyperspectral/band_selection/train.py:36]: Curious here about the "vanilla" naming?  And while I'm after nits, × (hex code d7) renders more universally.  (I am blushing as I lodge this comment, over it's persnicketyness.  Feel free to disregard.)

**Review by jpolchlo — APPROVED** [review]
Well played.  LGTM.

File: src/hyperspectral/hyperspectral/band_selection/models.py (hunk 1 above)
Comment (jamesmcclain) [reply src/hyperspectral/hyperspectral/band_selection/models.py:36]: Okay, that's a good idea.  I will move the code over there before merging :+1:

File: src/hyperspectral/hyperspectral/band_selection/train.py (hunk 2 above)
Comment (jamesmcclain) [reply src/hyperspectral/hyperspectral/band_selection/train.py:36]: This is to convey a sense of unadornedness, primitive simplicity, if you will.

**Pull request merged**